### PR TITLE
Credits burner

### PR DESCRIPTION
**Pull request by aaitor**
## Description

Add a description of your changes here.

## Is this PR related with an open issue?

Related to Issue #

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] Follows the code style of this project.
- [ ] Tests Cover Changes
- [ ] Documentation

### CODE REVIEW

**Review by eruizgar91 — APPROVED** [review]

File: conf/nginx/nginx.conf [hunk 1]
```diff
@@ -27,11 +27,27 @@ http {
...
+        '"user_id": "$user_id", '
+        '"request_time": "$request_time" }';
+
+
     sendfile        on;
 
     keepalive_timeout  65;
 
-    perl_set $env_introspection_url 'sub { return $ENV{"INTROSPECTION_URL"}; }';
+    #perl_set $env_introspection_url 'sub { return $ENV{"INTROSPECTION_URL"}; }';
```
Comment (r-marques) [inline conf/nginx/nginx.conf:50]: cleanup

File: src/credits-burner.ts [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+
+
+const burnTransactions = async (nvm: Nevermined, logs: any[], account: Account): Promise<TransactionsProcessed> => {
+  const results = []
+  const errors = []
+  let activeContractAddress = ''
+  
+
+  // const accounts = await nvm.accounts.list()
+  // const account = accounts[accountIndex]
```
Comment (r-marques) [inline src/credits-burner.ts:101]: cleanup

File: src/credits-burner.ts [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+
+  logger.debug(`Using account: ${account.getId()}`)
+
+  for await (const log of logs) {
+  
+  // logs.forEach(async (log) => {
```
Comment (r-marques) [inline src/credits-burner.ts:107]: cleanup

File: src/credits-burner.ts [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+    process.exit(1)
+  }
+  
+}
+
+const getAccount = async (config: ConfigEntry, nvm: Nevermined): Promise<Account> => {
+  
+    try {      
+      const account = await nvm.accounts.getAccount(config.nvm.neverminedNodeAddress)
+      // const account = (await nvm.accounts.list())[0]
```
Comment (r-marques) [inline src/credits-burner.ts:246]: cleanup

File: src/oauth-server.ts [hunk 5]
```diff
@@ -18,9 +18,15 @@ const counter = meter.createCounter(metricName, {
...
-const logger = require('pino')({
+const logger = require('pino')({  
   level: process.env.NODE_ENV === 'production' ? 'info' : 'debug',
 })
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+// const txCacheLogger = require('pino')({  
+//   level: 'info'
+//   },
+//   process.env.TX_DESTINATION || '/tmp/nv-oauth-server.log' 
+// )
```
Comment (r-marques) [inline src/oauth-server.ts:29]: cleanup

File: src/oauth-server.ts [hunk 6]
```diff
@@ -245,7 +256,7 @@ app.post('/introspect', async (req, res) => {
   }
 
   // Error response
-  res.writeHead(401)
+  // res.writeHead(401)
```
Comment (r-marques) [inline src/oauth-server.ts:259]: is this correct?

Comment (aaitor) [reply src/oauth-server.ts]: fixed

**Review by r-marques — APPROVED** [review]

**Pull request merged**